### PR TITLE
2019 Wikimedia Hackathon Service Pack

### PR DESCRIPTION
**Pull request by yarl**
I will add list of issues closed soon.

### CODE REVIEW

**Comment by atsirlin** [comment]
@yarl: Thanks! I am curious if there were any changes in the back-end after the hackathon, or the 'close campaign' button has been implemented without the actual possibility of closing the campaign.

**Comment by mahmoud** [comment]
@atsirlin no backend changes (from me) yet! I'll review this and then reply your email, @atsirlin, but while I'm here I might as well say thank you for working on this at the hackathon. I was planning on doing more, but I caught a bad cold. Glad you two still got a bunch done, great work! OK reviewing now...

**Comment by atsirlin** [comment]
@mahmoud: thank you too, it was a pleasure to meet Pawel and get a bit more familiar with the code! However, it would also be great if someone replies to the e-mail that I sent around after the hackathon...

File: client/src/components/vote/vote-single/vote-single.html [hunk 1]
```diff
@@ -138,33 +167,43 @@ <h3>
...
             </p>
           </div>
-          <md-button class="md-secondary md-icon-button"
-              ng-href="{{ '//commons.wikimedia.org/wiki/File:' + $ctrl.rating.current.entry.name + '#filehistory' }}"
-              target="_blank">
+          <md-button
+            class="md-secondary md-icon-button"
+            ng-href="{{
+              '//commons.wikimedia.org/wiki/File:' +
+                $ctrl.rating.current.entry.name +
```
Comment (mahmoud) [inline client/src/components/vote/vote-single/vote-single.html]: is this going to need escaping or is `entry.name` already url escaped?

**Comment by slaporte** [comment]
This is looking good to me.

File: client/src/components/vote/vote-single/vote-single.html (hunk 1 above)
Comment (slaporte) [reply client/src/components/vote/vote-single/vote-single.html]: I think we're getting `entry.name` straight from mediawiki, so it should be URL friendly. 

We should have a test for this, though—`$ctrl.rating.current.entry.name` makes an appearance above, and there is `encodeURIComponent(image.entry.name)` elsewhere...

**Comment by slaporte** [comment]
> However, it would also be great if someone replies to the e-mail that I sent around after the hackathon...

@atsirlin could you re-send, if those questions are still unanswered?

**Comment by atsirlin** [comment]
I think it's fine for now

**Pull request merged**